server/storage/backend/read_tx.go

```diff
@@ -83,13 +100,62 @@ func (baseReadTx *baseReadTx) UnsafeRange(bucketType Bucket, key, endKey []byte,
 		limit = math.MaxInt64
 	}
 	if limit > 1 && !bucketType.IsSafeRangeBucket() {
-		panic("do not use unsafeRange on non-keys bucket")
+		panic("do not use UnsafeRange on non-keys bucket")
 	}
-	keys, vals := baseReadTx.buf.Range(bucketType, key, endKey, limit)
-	if int64(len(keys)) == limit {
-		return keys, vals
+
+	// ConcurrentReadTxNoCopyMode
+	if baseReadTx.bufferNoCopy && endKey == nil {
+		// Single key search. Key exists either bolt db or read buffer
+		k1, v1 := baseReadTx.unsafeRangeDB(bucketType, key, endKey, limit)
+		if int64(len(k1)) == limit {
+			return k1, v1
+		}
+		k2, v2 := baseReadTx.unsafeRangeBuff(bucketType, key, endKey, limit-int64(len(k1)))
+		if len(k2) != 0 {
+			// If rate of buffRangeHitCounter/rangeTotalCounter is high, we should not use ConcurrentReadTxNoCopyMode
+			buffRangeHitCounter.Inc()
+		}
+		return append(k2, k1...), append(v2, v1...)
```

Comment:
for single key search, we will return one result nor no result. we don't need to use append here, right?